index.js

```diff
@@ -117,15 +117,21 @@ module.exports = class Transaction {
    * @return {Buffer}
    */
   hash (signature) {
+    const cpyRaw = this.raw.slice(0)
```

Comment:
I think we have to get a copy of the variable first, otherwise .hash() will be destructive and reset the current vrs values.